test/unit/data/events/scheme/get-submitted-events.test.js

```diff
@@ -0,0 +1,97 @@
+const { CATEGORY } = require('../../../../mocks/values/category')
+
+const { PAYMENT_SUBMITTED } = require('../../../../../app/constants/events')
+const { PAYMENT_EVENT } = require('../../../../../app/constants/event-types')
+
+const { stringifyEventData } = require('../../../../helpers/stringify-event-data')
+
+const { getSubmittedEvents } = require('../../../../../app/data/events/scheme-id/get-submitted-events')
+
+let extractedEvent
+let enrichedEvent
+let events
+
+describe('get events', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    extractedEvent = JSON.parse(JSON.stringify(require('../../../../mocks/events/extracted')))
+    enrichedEvent = JSON.parse(JSON.stringify(require('../../../../mocks/events/enriched')))
+
+    stringifyEventData(extractedEvent)
+    stringifyEventData(enrichedEvent)
+
+    events = [extractedEvent, enrichedEvent]
+
+    mockGetClient.mockReturnValue(mockTableClient)
+    mockListEntities.mockReturnValue(events)
+  })
+
+  test('should get payment client', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockGetClient).toHaveBeenCalledTimes(1)
+  })
+
+  test('should get payment client once', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockGetClient).toHaveBeenCalledTimes(1)
+  })
+
+  test('should get payment client with payment event type', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockGetClient).toHaveBeenCalledWith(PAYMENT_EVENT)
+  })
+
+  test('should get payment events once', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockListEntities).toHaveBeenCalledTimes(1)
+  })
+
+  test('should get payment events with correlation id category', async () => {
+    await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(mockListEntities).toHaveBeenCalledWith({ queryOptions: { filter: mockOdata`category eq 'correlationId' and type eq 'uk.gov.defra.ffc.pay.payment.submitted'` } })
+  })
+
+  test('should return all payment events', async () => {
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(result.length).toBe(2)
+  })
+
+  test('should convert event data to json', async () => {
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(result[0].data).toEqual(extractedEvent.data)
+    expect(result[1].data).toEqual(enrichedEvent.data)
+  })
+
+  test('should return an empty array if no events', async () => {
+    mockListEntities.mockReturnValue([])
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    expect(result.length).toBe(0)
+  })
+
+  // can this be tested because we are mocking the response from table storage?
+  test('should only return events that are of type submitted', async () => {
+    const result = await getSubmittedEvents(PARTITION_KEY, CATEGORY)
+    result.forEach(event => {
+      expect(event.type).toBe(PAYMENT_SUBMITTED)
+    })
+  })
```

Comment:
Not with mocks in unit tests.  These tests are more testing the interface for table storage is being used as expected with the correct query parameters and the responses are handled appropriately. 

The type of test you're describing would be a local integration test in the `integration/local` folder.   You could load some events to storage, then use the SDK without mocks to see if only the submitted events are returned.

Reply:
Integration test added.